projects/ADuCM360_demo_cn0411/src/CN0411.c

```diff
@@ -1288,7 +1302,24 @@ void CN0411_cmd_solution(uint8_t *args, struct cn0411_device *cn0411_dev)
 		cn0411_dev->solution.tds_factor = TDS_NACL;
 		printf("Solution set to NaCl.\n");
 	} else {
-		printf("Incorrect input value!\n");
+		index = 0;
+		str = strtok (arg, ",");
+		while (str != NULL) {
+			input_val = atof(str);
+			if(input_val < 0 || index > 1) {
+				printf("Incorrect input value!\n");
+				break;
+			} else if(index == 0) {
+				cn0411_dev->solution.temp_coeff = input_val;
+				printf("Temperature coefficient set to %.2f.\n",
+				       cn0411_dev->solution.temp_coeff);
+			} else {
+				cn0411_dev->solution.tds_factor = input_val;
+				printf("TDS factor set to %.2f.\n", cn0411_dev->solution.tds_factor);
```

Comment:
i guess you can add `break` statement here; even if there's more input here, we can ignore it after this point;

Reply:
that's also fine